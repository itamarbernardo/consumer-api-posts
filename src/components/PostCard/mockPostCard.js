export const mockPostCard = {
  title: 'Title 1',
  body: 'Body 1',
  id: '1',
  cover: 'img/img.png',
};
