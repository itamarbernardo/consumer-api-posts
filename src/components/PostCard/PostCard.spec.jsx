import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { mockPostCard } from './mockPostCard';

const mockPost = mockPostCard;

describe('<PostCard />', () => {
  it('should render PostCard correctly ', () => {
    // const { debug } = render(<PostCard post={mockPost}/>)
    // debug() //Mostra toda a tela construida -> Serve para testes snapshot -> tira uma foto de como ta construido a estrutura e se algo mudar, já sabe que tá errado
    render(<PostCard post={mockPost} />);

    expect(screen.getByRole('img', { name: /Title 1/i })) //a imagem existe
      .toHaveAttribute('src', mockPost.cover); //e contem esse atributo

    //Vamos usar heading do h1 até o h6
    expect(screen.getByRole('heading', { name: /Title 1/i })).toBeInTheDocument();
    //Testa o paragrafo -> Não conseguimos pegar paragrafo com o role
    expect(screen.getByText('Body 1')).toBeInTheDocument();
  });

  it('shoud match snapshot', () => {
    const { container } = render(<PostCard post={mockPost} />);
    expect(container.firstChild).toMatchSnapshot(); //Cria uma pasta com o snapshot pra comparar com o teste posteriormente
  });
});
