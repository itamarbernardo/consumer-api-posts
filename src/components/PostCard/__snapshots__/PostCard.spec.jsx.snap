// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> shoud match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Title 1
    </h1>
    <p>
      Body 1
    </p>
  </div>
</div>
`;
